src/main/java/uk/nhs/digital/nhsconnect/lab/results/outbound/queue/ObjectSerializer.java

```diff
@@ -0,0 +1,22 @@
+package uk.nhs.digital.nhsconnect.lab.results.outbound.queue;
+
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.lab.results.outbound.fhir.FhirParser;
+
+@Component
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+public class ObjectSerializer {
+    private final FhirParser fhirParser;
+
+    @SneakyThrows
+    public String serialize(Object object) {
+        if (object instanceof IBaseResource) {
+            return fhirParser.encodeToString((IBaseResource) object);
+        }
```

Comment:
Just a note for the future, java 16 is expected to introduce [Pattern Matching for instanceof](https://openjdk.java.net/jeps/394), which would let us do something like
```suggestion
        if (object instanceof IBaseResource resource) {
            return fhirParser.encodeToString(resource);
        }
```
Unlikely to be able to use it professionally though until after the next LTS release, 17 :(

Reply:
That looks nice! 👍